### ISSUE TITLE
`--compile=all` gets stuck

### ISSUE THREAD

**Issue by fduple** [JuliaLang/julia#47662]
**Minimal working example**
Finishes successfully  in less than 90 sec:
`julia -J ~/julia/julia-1.8.3/lib/julia/sys.so --output-o z.o -e ' '`

Does not finish even after several hours, with no warnings, error messages or output file created:
`julia -J ~/julia/julia-1.8.3/lib/julia/sys.so --output-o z.o -e ' ' --compile=all`

**Additional info:**
These pull requests are relevant: #47357 #46935
```
julia> versioninfo()
Julia Version 1.8.3
Commit 0434deb161e (2022-11-14 20:14 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 16 × 12th Gen Intel(R) Core(TM) i5-12600KF
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, goldmont)
  Threads: 1 on 16 virtual cores
Environment:
  LD_PRELOAD = /usr/lib/x86_64-linux-gnu/libstdc++.so.6
```
I installed Julia by downloading julia-1.8.3-linux-x86_64.tar.gz, gpg-verified it with julia-1.8.3-linux-x86_64.tar.gz.asc, untarred it into ~/julia/ and made a soft link to ~/julia/julia-1.8.3/bin/julia

**Comment by JeffBezanson** [comment]
Wow, there is a `codeinst` with a circular `next` pointer.

**Comment by JeffBezanson** [comment]
I think this is a backporting error, only on 1.8.x. Fix incoming.

**Comment by fduple** [comment]
Resolved; tested in Julia 1.8.4 and 1.8.5
